### PR TITLE
JS post-OOP

### PR DESCRIPTION
**Pull request by vakulinina**
### JS post-OOP

[Demo](https://vakulinina.github.io/a-tiny-JS-world/)
[Code base](https://github.com/vakulinina/a-tiny-JS-world)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

File: submissions/vakulinina/js-post-oop/index.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+class Human extends Inhabitant {
+   constructor(name, gender, saying) {
+      super('human', name, gender, 2, saying);
+      this.hands = 2;
+   }
+
+   introduce() {
+      return [this.species, this.name, this.gender, this.legs, this.hands, this.saying, this.getFriendsList(this.friends)].join('; ');
+   }
```
Comment (OleksiyRudenko) [inline submissions/vakulinina/js-post-oop/index.js:32]: I see why you refer all the props here.
What would you change in order to have a scalable and maintainable implementation if allowed to expose properties in arbitrary order?
Imagine we need to establish dozens of species, so most likely we do not want every species `introduce` list all props as any change in the list of properties required for various species will require code changes in multiple classes and will quickly become a nightmare.

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@vakulinina great job!
Just want to see if you're familiar with the concepts of ancestor's methods re-use.

File: submissions/vakulinina/js-post-oop/index.js (hunk 1 above)
Comment (vakulinina) [reply submissions/vakulinina/js-post-oop/index.js]: Thank you, I was struggling to come up with a neater solution, but failed 😄 
Hope new one is ok.

File: submissions/vakulinina/js-post-oop/index.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+class Human extends Inhabitant {
+   constructor(name, gender, saying) {
+      super('human', name, gender, 2, saying);
+      this.hands = 2;
+   }
+
+   getProperties() {
+      let properties = super.getProperties();
+      properties.splice(3, 0, this.hands);
```
Comment (OleksiyRudenko) [inline submissions/vakulinina/js-post-oop/index.js:36]: This approach relies on an order of props and may easily become a bug when hierarchy grows.

**Review by OleksiyRudenko — APPROVED** [review]
@vakulinina good job!
I see why did you struggle. It was a compromise between objects' design and consistency of the output.

Your approaches are valid for this particular task and you exposed quite a plenty of creativity. 

However we have to keep maintainability and scalability at all times. No doubt you've taken your learnings from this expereince.

File: submissions/vakulinina/js-post-oop/index.js (hunk 2 above)
Comment (vakulinina) [reply submissions/vakulinina/js-post-oop/index.js:36]: Note taken. Thank you!

**Pull request merged**